README.md

```diff
@@ -1,6 +1,75 @@
 # External Health Monitor
 
-This repo contains sidecar controller and agent for volume health monitoring.
+The External Health Monitor is part of Kubernetes implementation of [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec). It was introduced as an Alpha feature in Kubernetes v1.19.
+
+## Overview
+
+The [External Health Monitor](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1432-volume-health-monitor) is impelmeted as two components: `External Health Monitor Controller` and `External Health Monitor Agent`.
+
+- External Health Monitor Controller:
+  - The external health monitor controller will be deployed as a sidecar together with the CSI controller driver, similar to how the external-provisioner sidecar is deployed.
+  - Trigger controller RPC to check the health condition of the CSI volumes.
+  - The external controller sidecar will also watch for node failure events. This component can be enabled via a flag.
+
+- External Health Monitor Agent:
+  - The external health monitor agent will be deployed as a sidecar together with the CSI node driver on every Kubernetes worker node.
+  - Trigger node RPC to check volume's mounting conditions.
+  - Note that currently we do not have CSI support for local storage. When the support is available, we will implement relavant CSI monitoring interfaces as well.
+
+The External Health Monitor needs to invoke the following CSI interface to support
+
+- External Health Monitor Controller:
+  - ListVolumes
+  - ControllerGetVolume
+- External Health Monitor Agent:
+  - NodeGetVolumeStats
+
+## Driver Support
+
+Note that currently External Health Monitor doesn't have CSI support for local storage. Only CSI monitoring interfaces of Mock Driver has been implemented. We are planing to support CSI Host Path Driver later.
+
+## Usage
+
+External Health Monitor needs to be deployed with CSI driver.
+
+### Build && Push Image
+
+> Now, the image path and tag are incorrect in deploy/kubernetes directory. You need to build images of controller and agent by yourself until release is cut and official images are available.
```

Comment:
Please remove this sentence.  Since we'll be using the final released image, this will be incorrect only for a short time.